File: app/src/main/java/com/example/david/lists/common/buildlogic/FirebaseAuthModule.java
package com.example.david.lists.common.buildlogic;

import com.google.firebase.auth.FirebaseAuth;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
final class FirebaseAuthModule {
    @Singleton
    @Provides
    FirebaseAuth firebaseAuth() {
        return FirebaseAuth.getInstance();
    }
}

File: app/src/main/java/com/example/david/lists/common/buildlogic/FirebaseAuthModule.kt
package com.example.david.lists.common.buildlogic

import com.google.firebase.auth.FirebaseAuth
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class FirebaseAuthModule {
    @Singleton
    @Provides
    fun firebaseAuth(): FirebaseAuth {
        return FirebaseAuth.getInstance()
    }
}
